File: src/Client/inventory-track/src/app/models/dto/warehouse/warehouse-state-response-dto.ts
import { WarehouseTypeResponseDto } from './warehouse-type-response-dto';
import { InventoryItemResponseDto } from '../inventory-item/inventory-item-response-dto';

export class WarehouseStateResponseDto {
  id!: string;
  name!: string;
  type!: WarehouseTypeResponseDto;
  itemsCount!: number;
  location!: string;
  quantity!: number;
  inventoryItems!: InventoryItemResponseDto[];
}

File: src/Services/InventoryService/InventoryService.Application/DTOs/Response/Warehouse/WarehouseStateResponseDto.cs
﻿using InventoryService.Application.DTOs.Base;
using InventoryService.Application.DTOs.Response.InventoryItem;
using InventoryService.Application.DTOs.Response.WarehouseType;

namespace InventoryService.Application.DTOs.Response.Warehouse;

public class WarehouseStateResponseDto : BaseDto
{
    public string Name { get; set; }
    public WarehouseTypeResponseDto Type { get; set; }
    public int ItemsCount { get; set; }
    public string Location { get; set; }
    public long Quantity { get; set; }
    public List<InventoryItemResponseDto> InventoryItems { get; set; }
    public Guid ResponsiblePersonId { get; set; }
}